pdctl/command/config_command.go

```diff
@@ -68,50 +121,83 @@ func showConfigCommandFunc(cmd *cobra.Command, args []string) {
 	fmt.Println(r)
 }
 
+func showAllConfigCommandFunc(cmd *cobra.Command, args []string) {
+	r, err := doRequest(cmd, configPrefix, http.MethodGet)
+	if err != nil {
+		fmt.Printf("Failed to get config: %s", err)
+		return
+	}
+	fmt.Println(r)
+}
+
 func setConfigCommandFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 2 {
 		fmt.Println(cmd.UsageString())
 		return
 	}
+	var (
+		value interface{}
+		path  string
+	)
+	opt, val := args[0], args[1]
 
-	url := getAddressFromCmd(cmd, schedulePrefix)
-	var value interface{}
-	data := make(map[string]interface{})
+	f, ok := optionRel[opt]
+	if !ok {
+		fmt.Println("Failed to set config: unknow option")
+		return
+	}
+	switch f.kind {
+	case scheduleOpt:
+		path = schedulePrefix
+	case replicateOpt:
+		path = replicatePrefix
+	}
 
-	r, err := http.Get(url)
+	r, err := doRequest(cmd, path, http.MethodGet)
 	if err != nil {
-		fmt.Printf("Failed to set config:[%s]\n", err)
+		fmt.Printf("Failed to set config: %s", err)
 		return
 	}
-	if r.StatusCode != http.StatusOK {
-		printResponseError(r)
-		r.Body.Close()
+	data := make(map[string]interface{})
+	err = json.Unmarshal([]byte(r), &data)
+	if err != nil {
+		fmt.Printf("Failed to set config: %s", err)
 		return
 	}
-
-	json.NewDecoder(r.Body).Decode(&data)
-	r.Body.Close()
-	value, err = strconv.ParseFloat(args[1], 64)
-	if err != nil {
-		value = args[1]
+	switch k := f.typ; true {
```

Comment:
I prefer `switch f.type` here and lists all numeric types instead of `k <= reflect.Float64`.